pkg/spec/job.go

```diff
@@ -115,6 +115,12 @@ func (job *JobSpecMod) createKubernetesController() (*batch_v1.Job, error) {
 	if job.ActiveDeadlineSeconds != nil {
 		jobSpec.ActiveDeadlineSeconds = job.ActiveDeadlineSeconds
 	}
+	// if RestartPolicy is not set by user default it to 'OnFailure'
+	if jobSpec.Template.Spec.RestartPolicy == "" {
+		jobSpec.Template.Spec.RestartPolicy = api_v1.RestartPolicyOnFailure
+	} else if jobSpec.Template.Spec.RestartPolicy == api_v1.RestartPolicyAlways {
+		return nil, fmt.Errorf("the Job %q is invalid: restartPolicy: unsupported value: \"Always\": supported values: OnFailure, Never", job.Name)
```

Comment:
I don't think kedge should be checking this, whatever value the user sets for RestartPolicy should be used. If there is nothing, then OnFailure should be set.
I think we can lose the `else if` condition.

Reply:
This gives better UX to the user IMO, people can find that something is wrong even before they go on to deploy on cluster.

There are many instances in kedge we do this, like the whole population we do, we can let k8s fail but we do fail, this let's user know that something is wrong even before they go to running cluster.

Reply:
We will try to make sure we reduce errors on clusters as much as possible, we won't reach 100% but we can always try to atleast do static checks all by us

Reply:
I think auto population of names, fields is fine, but adding checks like these for some parts and not for others brings out inconsistency IMO.
Also, delegating as much as we can to the cluster and Kube reduces our maintenance overhead.

Duplicating behavior that the cluster takes care of anyway worries me in terms of inconsistency it might introduce :(

WDYT @kadel

Reply:
I don't have a strong opinion on this, to be honest. 
But I'm leaning towards having simple checks like this.

If it is possible we should make sure that Kedge doesn't generate invalid output.

Reply:
@kadel this is not invalid output. The fields and values are fine. My concern is that what happens in case where cluster version x accepts `RestartPolicy` as `Always` while cluster version y does not.

We should let the cluster do the OpenShift/Kubernetes validations, and we should be doing validations only for our schema. WDYT?

Reply:
>  this is not invalid output.

How so? It is valid Yaml yes, but invalid Kubernetes definitions.

>We should let the cluster do the OpenShift/Kubernetes validations, and we should be doing validations only for our schema. WDYT?

what do you mean by our schema? Our schema shouldn't allow to generate invalid Kubernets

Reply:
Okay, I could agree to keep this and move forward!
@surajssd can we move separate the validation and fix part to `Validate()` and `Fix()` methods for this.
`Fix()` can take the setting of RestartPolicy part, and `Validate()` can take the part which checks RestartPolicy as Always for a Job, should be okay then.

Reply:
Agreed, let's move this to `Validate`